### PR TITLE
[GEOT-6082] Add support in App-Schema for JDBC and Solr mutivalued properties

### PR DESCRIPTION
**Pull request by nmco**
Associated issue:
https://osgeo-org.atlassian.net/browse/GEOT-6082

Integrations test for SQL and documentation:
https://github.com/geoserver/geoserver/pull/3040

### CODE REVIEW

**Comment by aaime** [comment]
I've tried to have a look, but I don't understand anything about the module and quickly drowned into my ignorance....

**Comment by nmco** [comment]
@aaime indeed this is some advanced App-Schema usage. The feature chaining syntax support works well for features but not that well for simple values (strings, numbers, etc ...), and if we bring on non relational storages like Solr or MongoDB is hard to handle those mappings in an efficient \ user friendly way.

The approach here was:

1. allow data stores to contribute custom syntax
2. implement that custom syntax to Apache Solr, multivalued fields
3. implement that custom syntax for SQL based data stores

File: modules/extension/app-schema/app-schema/src/main/java/org/geotools/jdbc/WrappedFilterToSql.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.StringWriter;
+import java.io.Writer;
+import org.geotools.data.complex.FeatureTypeMapping;
+import org.geotools.data.jdbc.FilterToSQL;
+import org.geotools.data.jdbc.FilterToSQLException;
+import org.opengis.filter.Filter;
+import org.opengis.filter.expression.Expression;
+
+/** Wraps a filter to SQL encoder allowing us to write directly to the output stream. */
+public final class WrappedFilterToSql {
```
Comment (mbarto) [inline modules/extension/app-schema/app-schema/src/main/java/org/geotools/jdbc/WrappedFilterToSql.java]: Does it need to be final?

**Comment by mbarto** [comment]
@nmco it generally looks good to me (from my limited knowledge of app-schema, I didn't spot any evident issue). Is it possible to add more unit tests?

**Comment by nmco** [comment]
The integration tests for this are in GeoServer pull requests:
https://github.com/geoserver/geoserver/pull/3040

Perhaps you missed them Mauro ?

**Comment by mbarto** [comment]
You are right. This is the way it goes with app schema. No blockers from me

**Pull request merged**